src/shared/qr.ts

```diff
@@ -245,3 +254,28 @@ export const getNewOutbreakExposures = (
   const newOutbreakExposures = detectedExposures.filter(item => newIds.indexOf(item.outbreakId) > -1);
   return newOutbreakExposures;
 };
+
+/**
+ * Look at all outbreak/checkin matches and remove duplicate matches
+ * so the same checkin never results in more than one exposure
+ */
+export const deduplicateMatches = (allMatches: MatchData[]) => {
+  const deduplicationDict: MatchDeduplicationDict = {};
+  for (const match of allMatches) {
+    const timestampStr = match.timestamp.toString();
+    if (
+      Object.keys(deduplicationDict).indexOf(timestampStr) === -1 ||
+      deduplicationDict[timestampStr].maxOutbreakEndTimestamp < match.outbreakEvent.endTime ||
+      deduplicationDict[timestampStr].maxSeverity < match.outbreakEvent.severity
```

Comment:
replace these logic statements with named functions to make the code more readable